streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -68,18 +69,18 @@
     private boolean commitOffsetNeeded = false;
     private boolean transactionInFlight = false;
     private final Time time;
-    private final TaskMetrics metrics;
+    private final TaskMetrics taskMetrics;
```

Comment:
I'm not sure why we need to pass around `streamsMetricsImpl` from StreamThread to everywhere else now?

Reply:
Previously, we passed around the interface only to cast it back to `StreamsMetricsImpl` in line 80 of this file. All I did was make TaskMetrics declare that it really does want a `StreamsMetricsImpl` instead of a `StreamsMetrics` explicitly.

If we're going to cast it anyway, why not just use the type system?